Too bad, it didn't help for OCaml, we are still getting only

```
running eval kmeans
   with tool ocaml
Broken pipe (os error 32)
 outcome error
expected undefined
Error: Process completed with exit code 1.
```

so either this is an OS error that is not turned into an OCaml exception (my OCaml is very rusty, I don't know if that's likely; I guess it can happen if the binary is somehow wrong, compiled on a different architecture (or with different CPU extensions?)  than it runs on) or it happens outside of the main computation (`f input`). If the latter, maybe we could wrap *everything* in a `try...with` bracket. Or it may be an error in the test harness, not the main OCaml executable at all. No idea.

It may still make sense to merge the extra error reporting for OCaml and monitor in CI if it ever causes problems (so far, it doesn't seem so --- some OCaml tests passed fine).

So, again, the PR is ready for review.

--- a/tools/horde-ad/cabal.project
+++ b/tools/horde-ad/cabal.project
@@ -5,7 +5,9 @@ packages: gradbench.cabal
 -- The other options sound like plausible speedups looking at their documentation
 -- and apparently don't lower performance at least.
 package ox-arrays
-  flags: +nonportable-simd
+  -- CI seems to sometimes compile on a different CPU than the binary is run on
+  -- so this optimisation has to be disabled
+  -- flags: +nonportable-simd
   ghc-options: -optc-ffast-math -optc-fassociative-math -optc-fno-signed-zeros -optc-fno-trapping-math -optc-freciprocal-math
 
 ghc-options: -fexcess-precision

--- a/tools/ocaml/run/main.ml
+++ b/tools/ocaml/run/main.ml
@@ -110,6 +110,7 @@ let () =
           | Msg_Define (_id, mname) ->
              reply [("success", `Bool (List.exists (fun l -> fst (fst l) = mname) modules))]
           | Msg_Evaluate (_id, mname, fname, input) ->
+            try
              match look (mname, fname) modules with
                f ->
                try
@@ -122,6 +123,9 @@ let () =
                  let msg = Printexc.to_string e in
                  reply [("success", `Bool false);
                         ("error", `String (Printf.sprintf "tool exception:\n%s" msg))];
-                 raise e)
+                 raise e
+            with _ ->
+              reply [("success", `Bool false);
+                     ("error", `String (Printf.sprintf "unknown function: %s" fname))])
     | None -> exit 0
   done

